docs/Quick_start_guide.rst

```diff
@@ -0,0 +1,144 @@
+===========================
+Quick start guide
+===========================
+
+Usage
+-----
+
+Basic examples
+..............
+
+Here is a simple python script that reads an image, applies a median image filter (radius of 2 pixels), and writes the resulting image in a file.
+
+  ::
+
+    import itk
+
+    image = itk.imread("/path/to/my/image.png")
+    median = itk.MedianImageFilter.New(image, Radius = 2)
+    itk.imwrite(median, "path/to/my/output_image.png")
+
+There are two ways to instantiate filters with ITKPython:
+
+- Implicit (recommended): ITK type information is automatically detected from the data. Under the hood typed filter objects and images are created. 
+
+  ::
+
+    # Use `ImageFileReader` instead of the wrapping function `imread` to illustrate this example.
+    reader = itk.ImageFileReader.New(FileName="/path/to/my/image.png")
+    # Here we specify the filter input explicitly
+    median = itk.MedianImageFilter.New(Input=reader.GetOutput())
+    # Same as above but shortened. `Input` does not have to be specified.
+    median = itk.MedianImageFilter.New(reader.GetOutput())
+    # Same as above. `.GetOutput()` does not have to be specified.
+    median = itk.MedianImageFilter.New(reader)
+
+- Explicit: This can be useful if a filter cannot automatically select the type information (e.g. `CastImageFilter`), or to detect type mismatch errors which can lead to cryptic error messages.
+
+  - Explicit instantiation of median image filter
+
+  ::
+
+    ImageType = itk.Image[itk.F,2]
```

Comment:
That is a good comment. Since it is a type, more than a variable, I write them with uppercases to differentiate them. But indeed the python style is to have variables as lowercase. @thewtex What's your take on this style issue?